python/lsst/ctrl/mpexec/mpGraphExecutor.py

```diff
@@ -501,51 +511,48 @@ def _executeQuantaMP(self, graph, butler):
                     # finished
                     exitcode = job.process.exitcode
                     quantum_report = job.report()
-                    if quantum_report:
-                        self.report.quantaReports.append(quantum_report)
+                    self.report.quantaReports.append(quantum_report)
                     if exitcode == 0:
                         jobs.setJobState(job, JobState.FINISHED)
                         job.cleanup()
                         _LOG.debug("success: %s took %.3f seconds", job, time.time() - job.started)
                     else:
-                        self.report.status = ExecutionStatus.FAILURE
-                        # failMessage() has to be called before cleanup()
-                        message = job.failMessage()
-                        jobs.setJobState(job, JobState.FAILED)
+                        if job.terminated:
+                            # Was killed due to timeout.
+                            if self.report.status == ExecutionStatus.SUCCESS:
```

Comment:
If it is killed due to time out and reports success why can't that be treated as success? How can you be timed out and successful?

Reply:
It is treated as success, `terminated` is False for that case, and check for `exitcode == 0` is in the `if` branch above. This branch is only for the case when it was actually killed by a signal (had non-zero `exitcode`)